ERCS/erc-7798.md

```diff
@@ -0,0 +1,200 @@
+1. The merchant's device emits an NFC signal via host card emulation (HCE) for the customer’s device to read.
+1. The customer, running their wallet software of choice, reads the NFC signal. The customer’s wallet constructs the necessary transaction(s) to execute the onchain checkout. 
+1. The wallet then signs and submits all transactions.
+1. Upon successful onchain execution of these transactions, the checkout is complete.
+
+#### Use Case 2: Peer-to-peer Sends
+1. Alice requests a payment on her wallet app by emitting an NFC signal. The contents of the signal is an [ERC-681](erc-681.md) URI
+> []will this only work with 681? 681 has limitations, like receiver intent which would be great to solve too.
+
+1. [ERC-681](erc-681.md) URI Bob reads the NFC signal from Alice on his wallet app and constructs the transaction
+2. Bob holds his phone close to Alice's which triggers the NFC flow.
+1. Bob then signs and submits the transaction onchain.
+1. Send flow is complete.
+
+### requestContactlessPayment
+The requestContactlessPayment method will be added as a standard method to the [EIP-1193 Ethereum Provider JavaScript API](eip-1193). 
+
+This method takes in two parameters: `type` and `uri`.
+
+- `type` is an enum that represents the payload type:
+    - 1 means the URI is an ERC-681-compliant payload.
+    - 2 means the URI points to an HTTP relayer, where the response of the relayer is either a JSON object with the transaction data already encoded as a 0x string, or a message for the customer to sign. The customer wallet is responsible for constructing, signing and submitting the transaction/message.
+
+- `uri` represents either the ERC-681-compliant payload or the relayer endpoint to query for the JSON payload. 
+
+Example calls:
+
+```js 
+window.ethereum.request({ 
+  method: 'requestContactlessPayment', 
+  params: { 
+    type: 1,
+    uri: "ethereum:0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913@8453/transfer?address=0xf7d07Ee99095FDC09001710Ec232162a788BB989&uint256=1e5",
+});
+
+window.ethereum.request({ 
+  method: 'requestContactlessPayment', 
+  params: { 
+    type: 2,
+    uri: "https://nfcrelay.xyz/paymentTxParams?uuid=1234-abcd-56678",
+    verificationCode: "1234567890"
+}
+});
+```
+
+### Type 1
+
+The transaction is a regular crypto send. The receiving party transmits the [ERC-681](erc-681.md) URI to the sender’s wallet. 
+
+
+### Type 2
+
+Handles cases where the payload necessary to construct, sign, and submit the transaction needed for payment is too large to emit over NFC. The intended flow is as follows: 
+
+1. The merchant dapp makes a POST request to an NFC relayer endpoint with the information needed for completing a sale. 
+1. The merchant’s device then transmits the NFC relayer endpoint URI and verification code to the customer via any means
+1. The customer’s device then makes a GET request to the endpoint URI
+
+One of three payloads can be returned from the NFC relayer:
```

Comment:
What about a payload type where any arbitrary wallet RPC call can be included? Whether that's `eth_sendTransaction` or more recent specs such as `wallet_sendCalls` or `wallet_grantPermissions`.